### PR TITLE
Az 3.0 release

### PR DESCRIPTION
**Pull request by erich-wang**
## Description

Az 3.0 release candidate 1

## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

File: tools/StaticAnalysis/Exceptions/Az.IotHub/SignatureIssues.csv [hunk 1]
```diff
@@ -7,6 +7,7 @@
...
 "Microsoft.Azure.PowerShell.Cmdlets.IotHub.dll","Microsoft.Azure.Commands.Management.IotHub.SetAzureRmIotHub","Set-AzIotHub","1","8410","Parameter Units of cmdlet Set-AzIotHub does not follow the enforced naming convention of using a singular noun for a parameter name.","Consider using a singular noun for the parameter name."
 "Microsoft.Azure.PowerShell.Cmdlets.IotHub.dll","Microsoft.Azure.Commands.Management.IotHub.SetAzureRmIotHub","Set-AzIotHub","1","8410","Parameter EnableFileUploadNotifications of cmdlet Set-AzIotHub does not follow the enforced naming convention of using a singular noun for a parameter name.","Consider using a singular noun for the parameter name."
+"Microsoft.Azure.PowerShell.Cmdlets.IotHub.dll","Microsoft.Azure.Commands.Management.IotHub.SetAzureRmIotHub","Set-AzIotHub","1","8410","Parameter OperationsMonitoringProperties of cmdlet Set-AzIotHub does not follow the enforced naming convention of using a singular noun for a parameter name.","Consider using a singular noun for the parameter name."
```
Comment (erich-wang) [inline tools/StaticAnalysis/Exceptions/Az.IotHub/SignatureIssues.csv]: need to remove this added line

File: ChangeLog.md [hunk 2]
```diff
@@ -1,3 +1,168 @@
+## 3.0.0 - November 2019
+### General
+* Az.PrivateDns 1.0.0 released
+
+#### Az.Accounts
+* Add a deprecation message for 'Resolve-Error' alias.
+
+#### Az.Advisor
+* Added new category 'Operational Excellence' to Get-AzAdvisorRecommendation cmdlet.
+
```
Comment (markcowl) [inline ChangeLog.md]: I don't see any change for Az.Batch

File: src/Batch/Batch/Az.Batch.psd1 [hunk 3]
```diff
@@ -12,7 +12,7 @@
 # RootModule = ''
 
 # Version number of this module.
-ModuleVersion = '2.0.0'
+ModuleVersion = '2.0.1'
```
Comment (markcowl) [inline src/Batch/Batch/Az.Batch.psd1]: This was a breaking change release for Az.Batch

File: tools/StaticAnalysis/Exceptions/Az.ServiceFabric/BreakingChangeIssues.csv [hunk 4]
```diff
@@ -1,2 +1,4 @@
 "AssemblyFileName","ClassName","Target","Severity","ProblemId","Description","Remediation"
```
Comment (erich-wang) [inline tools/StaticAnalysis/Exceptions/Az.ServiceFabric/BreakingChangeIssues.csv]: all contents except title should be removed

File: src/FrontDoor/FrontDoor/Az.FrontDoor.psd1 [hunk 5]
```diff
@@ -53,7 +53,7 @@ DotNetFrameworkVersion = '4.7.2'
 # ProcessorArchitecture = ''
 
 # Modules that must be imported into the global environment prior to importing this module
-RequiredModules = @(@{ModuleName = 'Az.Accounts'; ModuleVersion = '1.6.3'; })
+RequiredModules = @(@{ModuleName = 'Az.Accounts'; ModuleVersion = '1.6.4'; })
 
 # Assemblies that must be loaded prior to importing this module
```
Comment (markcowl) [inline src/FrontDoor/FrontDoor/Az.FrontDoor.psd1]: No changelog change

File: ChangeLog.md (hunk 2 above)
Comment (erich-wang) [reply ChangeLog.md]: good catch, will add it back

File: src/Batch/Batch/Az.Batch.psd1 (hunk 3 above)
Comment (erich-wang) [reply src/Batch/Batch/Az.Batch.psd1]: Yes, the preview version is 2.0.0-preview, so I manually change to 2.0.1

Comment (markcowl) [reply src/Batch/Batch/Az.Batch.psd1]: Yeah, my guess is that the mismatch is what prevented the update of az.psd1 and for recovery services

File: src/Resources/Resources/ChangeLog.md [hunk 6]
```diff
@@ -19,6 +19,10 @@
 -->
 ## Upcoming Release
 
+## Version 1.7.1
+
+* Update dependency assemlby Microsoft.Extensions.Caching.Memory from 1.1.1 to 2.2
```
Comment (markcowl) [inline src/Resources/Resources/ChangeLog.md]:
```suggestion
* Update dependency assembly Microsoft.Extensions.Caching.Memory from 1.1.1 to 2.2
```

Comment (markcowl) [reply src/Resources/Resources/ChangeLog.md]: Note - thsi is a minor type, I would not hold up the release for this, but if we merge any further PRs, we could fix this

**Pull request merged**